source/Symbol/Symtab.cpp

```diff
@@ -283,67 +322,18 @@ void Symtab::InitNameIndexes() {
-            entry.cstring = ConstString(cxx_method.GetBasename());
-            if (entry.cstring) {
-              // ConstString objects permanently store the string in the pool
-              // so calling GetCString() on the value gets us a const char *
-              // that will never go away
-              const char *const_context =
-                  ConstString(cxx_method.GetContext()).GetCString();
-
-              if (!const_context || const_context[0] == 0) {
-                // No context for this function so this has to be a basename
-                m_basename_to_index.Append(entry);
-                // If there is no context (no namespaces or class scopes that
-                // come before the function name) then this also could be a
-                // fullname.
-                m_name_to_index.Append(entry);
-              } else {
-                entry_ref = entry.cstring.GetStringRef();
-                if (entry_ref[0] == '~' ||
-                    !cxx_method.GetQualifiers().empty()) {
-                  // The first character of the demangled basename is '~' which
-                  // means we have a class destructor. We can use this
-                  // information to help us know what is a class and what
-                  // isn't.
-                  if (class_contexts.find(const_context) == class_contexts.end())
-                    class_contexts.insert(const_context);
-                  m_method_to_index.Append(entry);
-                } else {
-                  if (class_contexts.find(const_context) !=
-                      class_contexts.end()) {
-                    // The current decl context is in our "class_contexts"
-                    // which means this is a method on a class
-                    m_method_to_index.Append(entry);
-                  } else {
-                    // We don't know if this is a function basename or a
-                    // method, so put it into a temporary collection so once we
-                    // are done we can look in class_contexts to see if each
-                    // entry is a class or just a function and will put any
-                    // remaining items into m_method_to_index or
-                    // m_basename_to_index as needed
-                    mangled_name_to_index.Append(entry);
-                    symbol_contexts[entry.value] = const_context;
-                  }
-                }
-              }
+        const SymbolType type = symbol->GetType();
+        if (type == eSymbolTypeCode || type == eSymbolTypeResolver) {
+          // Other schemes are not relevant in the Swift use case.
+          bool is_relevant_itanium = !lldb_skip_name(
+              entry.cstring.GetStringRef(), Mangled::eManglingSchemeItanium);
+
+          if (is_relevant_itanium) {
+            if (mangled.DemangleWithRichManglingInfo(rmc, lldb_skip_name)) {
+              RegisterMangledNameEntry(entry, class_contexts, backlog, rmc);
```

Comment:
Had a merge conflict here due to logic changes

Reply:
Why did you add the is_relevant_itanium bool that AFAICS is not present upstream? doesn't the lldb_skip_name argument to DemangleWithRichManglingInfo fulfill the same purpose? If not, why is this condition not on llvm.org?

I have merged your patch to upstream-with-swift a couple days ago, and this is the resolution I came up with:
```
        const SymbolType type = symbol->GetType();
        if (type == eSymbolTypeCode || type == eSymbolTypeResolver) {
          if (mangled.DemangleWithRichManglingInfo(rmc, lldb_skip_name))
            RegisterMangledNameEntry(entry, class_contexts, backlog, rmc);
          else if (SwiftLanguageRuntime::IsSwiftMangledName(name.str().c_str())) {
            lldb_private::ConstString basename;
            bool is_method = false;
            ConstString mangled_name = mangled.GetMangledName();
            if (SwiftLanguageRuntime::MethodName::
                    ExtractFunctionBasenameFromMangled(mangled_name, basename,
                                                       is_method)) {
              if (basename && basename != mangled_name) {
                entry.cstring = basename;
                if (is_method)
                  m_method_to_index.Append(entry);
                else
                  m_basename_to_index.Append(entry);
              }
            }
          }
        }
```

Is this wrong?

Reply:
Btw. the other way around is not worrying, as Swift names have a `_T` prefix while Itanium uses `_Z`. Thus, IPD will never take on a Swift name and will always reach `SwiftLanguageRuntime::IsSwiftMangledName()`.